src/Features/Core/Portable/Diagnostics/CodeAnalysisDiagnosticAnalyzerService.cs

```diff
@@ -103,6 +121,58 @@ public async ValueTask RunAnalysisAsync(Project project, CancellationToken cance
             // TODO: Below call will eventually be replaced with a special workspace refresh request that skips pulling
             // document diagnostics and also does not add any delay for pulling workspace diagnostics.
             _diagnosticAnalyzerService.RequestDiagnosticRefresh();
+
+            bool FilterAnalyzer(DiagnosticAnalyzer analyzer)
```

Comment:
this is a move from teh diag analzyer service itself.   i want to take it out of the job of figuring these things out.